torch_xla/experimental/distributed_checkpoint/manager.py

```diff
@@ -77,14 +97,48 @@ def __init__(self,
             Default: 1, which only allows a single async checkpoint to be
             pending at a time.
     """
-    raise NotImplementedError
+    assert dist.is_initialized(), "A process group is required."
+    assert save_period > 0, "save_period must be positive"
+    assert async_queue_size > 0, "async_queue_size must be positive"
+    assert max_to_keep >= 0, "max_to_keep must be non-negative"
+
+    self.base_path = path
+    self.save_period = save_period
+    self.max_to_keep = max_to_keep
+
+    # Cache tracked steps in a heap for efficient clearing.
+    self._tracked_steps = self._load_tracked_steps()
```

Comment:
Why a heap? Can the .save be called out of order?

Reply:
It's sort of an edge case, but if the user restores from an earlier checkpoint and changes the save_period, the steps will be out of order. Using a heap ensures that we always delete the lowest step number.

Reply:
heap is still o(logn). If sort it after loading, then the later deletion from queue is just o(1)

Reply:
I'll update to delete the oldest by timestamp instead of lowest step number.